cli/src/cli_util.rs

```diff
@@ -3847,6 +3848,18 @@ fn map_clap_cli_error(mut cmd_err: CommandError, ui: &Ui, config: &StackedConfig
     let Some(err) = cmd_err.error.downcast_ref::<clap::Error>() else {
         return cmd_err;
     };
+    if let Some(ContextValue::String(cmd)) = err.get(ContextKind::InvalidSubcommand) {
+        let cmd = cmd.as_str();
+        if matches!(cmd, "clone" | "init") {
+            // git commands
+            return cmd_err.clone().hinted(formatdoc!(
+                r#"
+                   Perhaps you meant `jj git {cmd}`?
+                   You can configure `aliases.{cmd}=["git", "{cmd}"]` if
+                   you want `jj {cmd}` to work and always use the Git backend."#
```

Comment:
nit: I wouldn't insert `\n` after `if`. If you like, these two can be rendered as separate hints.

Reply:
Oh, I hadn't realized we can do multiple hints. Nice!